### ISSUE TITLE
adding jalali calendar system to datePicker[Feature request]

### ISSUE THREAD

**Issue by dev-am1** [mengxiong10/vue2-datepicker#507]
**What problem does this feature solve?**
it helps Persian users with picking time in their native calendar system.

**What does the proposed API look like?**
it looks like the same as the original but with different values . 

hi I'm a nextcloud user,
is there a way to show jalali calendar based on Gregorian calendar, just like locale options for month and day names, 
I know this date picker must work with Gregorian calendar system but can it show the days ( current day) and year in jalali calendar system? ( just showing it, not actually representing the date)

![datePicker](https://user-images.githubusercontent.com/59691125/89355545-e020a980-d6d0-11ea-8286-344e9133e7c1.png)

if you could help me with where I must begin to look for fixing this in the index.esm.js (node module), I will be extremely thankful.

**Comment by mengxiong10** [comment]
I think you can use a computed value.
```html
<date-picker  v-model="dateWhichShouldShow" />

```
```js
data() {
  return {
    actualDate: null,
  };
},
computed: {
  dateWhichShouldShow: {
    get() {
      const date = new Date(this.actualDate);
      // ... transfrom the date to what you want show;
      return date;
    },
    set(val) {
      const value = new Date(val);

      this.actualDate = ...
      // ... transfrom the date to actual date
    },
  },
},


```

**Comment by zumoshi** [comment]
There aren't the same number of days in Jalali months compared to the Gregorian system. Simply changing the value after it is selected in Gregorian won't solve the issue. Most users won't know what day it is in the foreign calendar system off hand.
There is a fork of this library that has switched the calendar: https://github.com/MostafaTech/vue-datepicker-persian/
But it would be nice if the main branch supported both systems and let us choose it via a prop for international websites.

**Comment by mengxiong10** [comment]
@zumoshi 
Thank you for the provider’s project.
I don’t know much about these calendar systems.
I can find out and then make a way to support them.

**Comment by zumoshi** [comment]
Thank you for your interest in adding support. Most of the libraries I've seen support only one of the systems forcing us to have two separate libraries if we have to support both.

But I will provide some links here for now:

- [The calendar system](https://en.wikipedia.org/wiki/Solar_Hijri_calendar)
    - Day per month: `[31, 31, 31, 31, 31, 31, 30, 30, 30, 30, 30, 29 or 30]`
    - Month names: `["فروردین", "اردیبهشت", "خرداد", "تیر", "امرداد", "شهریور", "مهر", "آبان", "آذر", "دی", "بهمن", "اسفند"]`
    - Weekday names: `['شنبه', 'یکشنبه', 'دوشنبه', 'سه شنبه', 'چهار شنبه', 'پنج\u200cشنبه', 'جمعه']`
    - (P.s don't read these left-to-right, they are valid js objects, copy in chrome console and use dropdown to see correct order)
- Converting between Jalali/Gregorian:
    - [date-fns-jalali](https://github.com/smmoosavi/date-fns-jalali) (low stars but same library you are already using)
    - [moment-jalaali](https://github.com/jalaali/moment-jalaali) (better maintained)
    - [PersianDate](https://github.com/babakhani/PersianDate) (light weight, only converting with no extra tools)
- Demo of date pickers in action:
    - [vue2-persian-datepicker](https://dyonir.github.io/vue2-persian-datepicker/)
    - [vue-persian-datetime-picker](https://talkhabi.github.io/vue-persian-datetime-picker/#/)
    - [pwt.datepicker](http://babakhani.github.io/PersianWebToolkit/doc/datepicker/example/)
- Nice to have features:
    - Using timestamp/gregorian date in v-model but displaying jalali/shamsi. Since most databases can't store it directly.
    - Showing Persian numerals (e.g. `۰۱۲۳۴۵۶۷۸۹` vs `012345679`)
    - Exposing jalali<->georgian conversion functions used internally, so we wouldn't have to install yet another library conversion was needed elsewhere.

Feel free to ask me any questions if you need clarification. And I'm also available if you need any translations to Persian for the ui.

**Comment by dev-am1** [comment]
I've fixed this issue in calendar 2.0.3 and datepicker with forced locale on 'fa', it is fully functional and doesn't produce any errors,
it can successfully sync with ubuntu's calendar as well . 

![Screenshot_2020-11-07 آبان 1399 - تقویم - حافظه ابری ویرا نسخه 0 1 3](https://user-images.githubusercontent.com/59691125/98425182-6cb07e80-20a9-11eb-8166-d3ae1352b23b.png)

you can check my github for fully ready to use builded minified source code, or unbuilded one and off course the rtl and fully persian nextcloud projects.
